core/src/main/java/nl/nn/adapterframework/filesystem/ExchangeFileSystemCache.java

```diff
@@ -0,0 +1,157 @@
+/*
+   Copyright 2022 WeAreFrank!
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+package nl.nn.adapterframework.filesystem;
+
+import microsoft.exchange.webservices.data.core.service.folder.Folder;
+import microsoft.exchange.webservices.data.property.complex.FolderId;
+import nl.nn.adapterframework.util.LogUtil;
+import org.apache.logging.log4j.Logger;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Acts as a cache for EWS components.
+ * Stores Folder objects in memory for quick access.
+ *
+ * @author Laurens Mäkel.
+ */
+public class ExchangeFileSystemCache {
+	private Logger log = LogUtil.getLogger(this);
+
+	private final Map<String, FolderId> baseFolders = new HashMap<>(); // <Mailbox+FolderNaam, FolderId>
```

Comment:
comment klopt niet, index is alleen maar mailbox

Reply:
Correct, aangepast.

Reply:
Je hebt nu de verkeerde aangepast.